src/mpl/configure.ac

```diff
@@ -1011,6 +1011,23 @@ if test "$have_gpu" = "no" ; then
     fi
 fi
 AM_CONDITIONAL([MPL_HAVE_ZE],[test "X${pac_have_ze}" = "Xyes"])
+
+# Check HIP availability
+if test "$have_gpu" = "no" ; then
+        PAC_CHECK_HEADER_LIB_OPTIONAL([hip],[hip_runtime_api.h],[hiprt],[hipStreamSynchronize])
```

Comment:
I think we can wrap this one with
```
PAC_PUSH_FLAG(CPPFLAGS)
PAC_APPEND_FLAG([-D__HIP_PLATFORM_AMD__], [CPPFLAGS])
PAC_CHECK_HEADER_LIB_OPTIONAL(...)
PAC_POP_FLAG(CPPFLAGS)
```
This would temporarily add the `__HIP_PLATFROM_AMD__` definition when checking the availability of HIP.

And we can do `PAC_APPEND_FLAG(...)` again later when HIP is actually detected (after AC_DEFINE).

Reply:
@yfguo just seeing these comments, I just pushed few commits before I saw these comments, the code is built and the tests (few P2P tests) on an AMD GPU were passed.